### ISSUE TITLE
Citeproc (and/or pandoc) hangs in specific sequence of three citations with a space before a colon

### ISSUE THREAD

**Issue by htdebeer** [jgm/pandoc#7060]
I regenerated my website today and pandoc now hangs on certain documents with citations. I have no idea when this behavior was introduced, but I have not updated these files in years. I have been able to create a minimal example:

Pandoc hangs when I run `pandoc --from markdown --to html --citeproc --csl chicago-fullnote-bibliography.txt --bibliography testbib.bib -o test.html test.md`. 

The contents of these files are:

- markdown input file `test.md`:
  
  ```markdown
  [@Sammet1969a, pp. 3–4]
  [@Backus1978a, p. 165]
  [@Sammet1969a, p. 134]
  ```
  
  When I remove one of these three citations, or reorder them, the issue does no longer occur.
- bibliography bibtex file `testbib.bib`:

  ```bibtex
  @inproceedings{ Backus1978a,
	author = {John Backus},
	title = {The history of FORTRAN I, II, and III},
	booktitle = {HOPL-1: The first ACM SIGPLAN conference on History of programming languages},
	year = {1978},
	pages = {165--180},
	location = {Los Angeles, CA},
	doi = {10.1145/800025.808380},
	publisher = {ACM Press},
	address = {New York, NY, USA}
  }

  @book{ Sammet1969a,
	author = {Jean E. Sammet},
	title = {Programming languages : history and fundamentals},
	publisher = {Prentice-Hall},
	address = {Englewood Cliffs, N. J.},
	year = {1969},
	series = {Series in Automatic Computation}
  }
  ```
  
  If I remove the space before the colon in the Sammet's title, the issue disappears as well!
  
- CSL file is `chicago-fullnote-bibliography.txt` (attached as TXT file because CSL not accepted by GitHub: 
[chicago-fullnote-bibliography.txt](https://github.com/jgm/pandoc/files/5894608/chicago-fullnote-bibliography.txt)). I downloaded also the latest version of this CSL style, but with the same effect. However, if I use `apa.csl` instead, the issue does not occur.

This happens with pandoc versions **2.11.2** and **2.11.4**

**Comment by htdebeer** [comment]
I found a bibtex entry that does not work with the `apa.csl` anymore:

```bibtex
@InProceedings{Cole2014a,
  Title                    = {{Toward an argumentative grammar for socio-cultural/cultural-historical activity approaches to design research}},
  Author                   = {Cole, Michael and Engestr{\"o}m, Yrj{\"o} and Sannino, Annalisa and Guti{\'e}rrez, Kris and Jurow, Susan and Packer, Martin and Penuel, William R. and Johnson, Raymond and Severance, Samuel and Leary, Heather and Miller, Susan},
  Booktitle                = {{ICLS 2014 Proceedings }},
  Year                     = {2014},
  Pages                    = {1254--1263}
}
```

Here I can resolve the issue by removing the space at the end of the booktitle attribute.

**Comment by jgm** [comment]
I can reproduce the issue. Oddly, it seems specific to the CSL style you're using.  (I also tried with chicago-note-bibliography and it hung, but it did not hang with chicago-author-date or another note style, modern-humanities-research-association.csl!)

Please submit the other issue (about the bibtex entry) as a separate issue because it seems unrelated.

**Comment by jgm** [comment]
Note that with the bibtex entry, pandoc hangs when using `pandoc -f biblatex -t csljson` but succeeds with `pandoc -f bibtex -t csljson`. Please include this information in the new issue.

**Comment by jgm** [comment]
My first guess from the clues you've given is that this may have something to do with the code for doing titlecase transformations.

**Comment by htdebeer** [comment]
Thanks for your quick replies!

The workaround of removing the superfluous space works great. Plus, as a result, my BibTeX files are now more neat :-)

I think these unfortunate formatting issues have occurred because I copied BibTeX entries from Google Scholar or publishers' websites. I also used JabRef and KBibTex, both of which do some BibTeX reformatting, but it seems they do not check or fix for these kind of issues.
